### PR TITLE
doc: Sync classref with current source

### PR DESCRIPTION
**Pull request by akien-mga**
Needs review/amends, some of those changes might not be fully intended.

### CODE REVIEW

File: doc/classes/ArrayMesh.xml [hunk 1]
```diff
@@ -272,14 +298,44 @@
...
 		</constant>
-		<constant name="ARRAY_FORMAT_WEIGHTS" value="128" enum="ArrayFormat">
+		<constant name="ARRAY_FORMAT_WEIGHTS" value="2048" enum="ArrayFormat">
 			Array format will include bone weights.
 		</constant>
-		<constant name="ARRAY_FORMAT_INDEX" value="256" enum="ArrayFormat">
+		<constant name="ARRAY_FORMAT_INDEX" value="4096" enum="ArrayFormat">
 			Index array will be used.
 		</constant>
+		<constant name="ARRAY_FORMAT_BLEND_SHAPE_MASK" value="-8185" enum="ArrayFormat">
```
Comment (akien-mga) [inline doc/classes/ArrayMesh.xml:318]: @reduz Overflowing `int32_t`?

File: doc/classes/ArrayMesh.xml [hunk 2]
```diff
@@ -272,14 +298,44 @@
...
+		<constant name="ARRAY_FORMAT_CUSTOM_MASK" value="7" enum="ArrayFormat">
+		</constant>
+		<constant name="ARRAY_COMPRESS_FLAGS_BASE" value="25" enum="ArrayFormat">
+		</constant>
+		<constant name="ARRAY_FLAG_USE_2D_VERTICES" value="33554432" enum="ArrayFormat">
+		</constant>
+		<constant name="ARRAY_FLAG_USE_DYNAMIC_UPDATE" value="67108864" enum="ArrayFormat">
+		</constant>
+		<constant name="ARRAY_FLAG_USE_8_BONE_WEIGHTS" value="134217728" enum="ArrayFormat">
+		</constant>
```
Comment (akien-mga) [inline doc/classes/ArrayMesh.xml:339]: All these values are pretty weird. Is that intended?

File: doc/classes/Control.xml [hunk 3]
```diff
@@ -133,6 +133,14 @@
 				Return [code]Array[/code] of [code]Vector2i[/code] text ranges, in the left-to-right order. Ranges should cover full source [code]text[/code] without overlaps. BiDi algorithm will be used on each range separately.
 			</description>
 		</method>
+		<method name="_unhandled_key_input" qualifiers="virtual">
```
Comment (akien-mga) [inline doc/classes/Control.xml:136]: That's already bound in `Node`, why is it now re-bound in `Control` (and with wrong type hint)?

Comes from efe5c250d5c @EricEzaM.

File: doc/classes/ArrayMesh.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/ArrayMesh.xml:318]: Same stuff on `Mesh`. BTW shouldn't we bind those only in `Mesh` and not `ArrayMesh`? Aren't they exactly the same?

File: doc/classes/RenderingServer.xml [hunk 4]
```diff
@@ -3084,16 +3068,16 @@
 		<constant name="ARRAY_TEX_UV2" value="5" enum="ArrayType">
 			Array is an UV coordinates array for the second UV coordinates.
 		</constant>
-		<constant name="ARRAY_BONES" value="6" enum="ArrayType">
+		<constant name="ARRAY_BONES" value="10" enum="ArrayType">
```
Comment (akien-mga) [inline doc/classes/RenderingServer.xml:3071]: Missing intermediate bindings?

File: doc/classes/RenderingServer.xml [hunk 5]
```diff
@@ -3114,40 +3098,19 @@
 		<constant name="ARRAY_FORMAT_TEX_UV2" value="32" enum="ArrayFormat">
 			Flag used to mark an UV coordinates array for the second UV coordinates.
 		</constant>
-		<constant name="ARRAY_FORMAT_BONES" value="64" enum="ArrayFormat">
+		<constant name="ARRAY_FORMAT_BONES" value="1024" enum="ArrayFormat">
```
Comment (akien-mga) [inline doc/classes/RenderingServer.xml:3101]: Missing bindings.

File: doc/classes/RichTextLabel.xml [hunk 6]
```diff
@@ -329,9 +329,6 @@
 			The raw text of the label.
 			When set, clears the tag stack and adds a raw text tag to the top of it. Does not parse BBCodes. Does not modify [member bbcode_text].
 		</member>
-		<member name="text_direction" type="int" setter="set_text_direction" getter="get_text_direction" enum="Control.TextDirection" default="0">
-			Base text writing direction.
-		</member>
```
Comment (akien-mga) [inline doc/classes/RichTextLabel.xml:334]: I guess this was a rebase mistake and is actually added by #42595 @bruvzg?

File: doc/classes/SurfaceTool.xml [hunk 7]
```diff
@@ -5,14 +5,15 @@
 	</brief_description>
 	<description>
 		The [SurfaceTool] is used to construct a [Mesh] by specifying vertex attributes individually. It can be used to construct a [Mesh] from a script. All properties except indices need to be added before calling [method add_vertex]. For example, to add vertex colors and UVs:
+		[b]FIXME: Code no longer valid in 4.0, add_uv and add_color were removed. What's the replacement?[/b]
```
Comment (akien-mga) [inline doc/classes/SurfaceTool.xml:8]: Should we just replace the `add_` methods by `set_` methods now? @reduz

File: doc/classes/SurfaceTool.xml [hunk 8]
```diff
@@ -167,11 +104,11 @@
 			</return>
 			<argument index="0" name="existing" type="ArrayMesh" default="null">
 			</argument>
-			<argument index="1" name="flags" type="int" default="31744">
+			<argument index="1" name="flags" type="int" default="0">
 			</argument>
 			<description>
 				Returns a constructed [ArrayMesh] from current information passed in. If an existing [ArrayMesh] is passed in as an argument, will add an extra surface to the existing [ArrayMesh].
-				Default flag is [constant Mesh.ARRAY_COMPRESS_DEFAULT]. See [code]ARRAY_COMPRESS_*[/code] constants in [enum Mesh.ArrayFormat] for other flags.
+				See [code]ARRAY_COMPRESS_*[/code] constants in [enum Mesh.ArrayFormat] for possible flags.
```
Comment (akien-mga) [inline doc/classes/SurfaceTool.xml:111]: What should this be updated to? There's only one `ARRAY_COMPRESS_*` flag left.

File: doc/classes/Control.xml (hunk 3 above)
Comment (EricEzaM) [reply doc/classes/Control.xml:136]: First guess would be that I didn't bind it as virtual? In mobile can't check atm

Comment (akien-mga) [reply doc/classes/Control.xml:136]: Why does it need to be bound though? It's already bound in Node, and Control inherits Node.

Comment (EricEzaM) [reply doc/classes/Control.xml:136]: Yeah that too lol

Comment (EricEzaM) [reply doc/classes/Control.xml:136]: I'll fix it up when I get to my computer

Comment (akien-mga) [reply doc/classes/Control.xml]: Fixed.

File: doc/classes/RenderingServer.xml (hunk 4 above)
Comment (akien-mga) [reply doc/classes/RenderingServer.xml:3079]: Fixed. You systematically forget to bind enum values in `rasterizer_server.cpp` :)

File: doc/classes/RenderingServer.xml (hunk 5 above)
Comment (akien-mga) [reply doc/classes/RenderingServer.xml:3117]: Fixed.

File: doc/classes/SurfaceTool.xml (hunk 7 above)
Comment (akien-mga) [reply doc/classes/SurfaceTool.xml]: Did that.

File: doc/classes/RichTextLabel.xml (hunk 6 above)
Comment (bruvzg) [reply doc/classes/RichTextLabel.xml:334]: Yes, it should be in #42595.

**Comment by akien-mga** [comment]
Removed the redundant bindings in `ArrayMesh` (already in `Mesh`) as advised by @reduz.

File: doc/classes/ArrayMesh.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/ArrayMesh.xml]: > Overflowing `int32_t`?

reduz will fix that in a follow-up PR.

> BTW shouldn't we bind those only in Mesh and not ArrayMesh? Aren't they exactly the same?

I fixed that.

File: doc/classes/ArrayMesh.xml (hunk 2 above)
Comment (akien-mga) [reply doc/classes/ArrayMesh.xml]: Seems intended. I'm not sure the `_BASE` variables are meant to be exposed though.

File: doc/classes/SurfaceTool.xml (hunk 8 above)
Comment (akien-mga) [reply doc/classes/SurfaceTool.xml]: Added a FIXME.

**Comment by aaronfranke** [comment]
> @akien-mga: Removed the redundant bindings in `ArrayMesh` (already in `Mesh`) as advised by @reduz.

See also #36382, specifically commit "Rebind Mesh/ArrayMesh enums" which removes the redundant bindings in `ArrayMesh`, but unlike here, it combines the docs by keeping the added detail currently present in the `ArrayMesh` docs.

**Comment by akien-mga** [comment]
> See also #36382, specifically commit "Rebind Mesh/ArrayMesh enums" which removes the redundant bindings in ArrayMesh, but unlike here, it combines the docs by keeping the added detail currently present in the ArrayMesh docs.

Seems good, though this commit also removes more duplicated enum bindings in ArrayMesh, and fixes the code examples accordingly. So if that's OK with you I'd propose to merge this PR and you can include the additional BlendShapeMode and documentation changes while rebasing #36382?

**Pull request merged**